include/model.php

```diff
@@ -702,34 +702,25 @@ public function filter_language_terms_orderby( $orderby, $args, $taxonomies ) {
 	 * @phpstan-return list<PLL_Language>
 	 */
 	protected function get_languages_from_taxonomies() {
-		$terms_by_type = $this->get_language_terms_by_type();
 
-		if ( empty( $terms_by_type['post'] ) || empty( $terms_by_type['term'] ) ) {
-			// `post` and `term` languages are mandatory.
-			return array();
-		}
+		// To get language taxonomies first.
+		$reversed_terms = array_reverse( $this->get_language_terms() );
 
-		$languages = array();
+		$terms_by_slug = array();
 
-		foreach ( $terms_by_type['post'] as $term_slug => $lang_term ) {
-			if ( ! isset( $terms_by_type['term'][ "pll_{$term_slug}" ] ) ) {
-				// A corresponding `term` language is mandatory.
-				continue;
+		foreach ( $reversed_terms as $term ) {
+			if ( $term instanceof WP_Term ) {
+				$key = 'language' === $term->taxonomy ? $term->slug : substr( $term->slug, 4 );
```

Comment:
Today I see why using `substr()`, but not sure I'll still in the future. Could you add a comment on why?

Reply:
All the language taxonomy slugs except the main one (language) are prefixed with `pll_`

See in 3.3 release https://github.com/polylang/polylang/blob/3.3.x/admin/admin-model.php#L47
and now in the future 3.4 release
https://github.com/polylang/polylang/blob/3.4-dev/admin/admin-model.php#L242

So to have the same key (language slug) between the language taxonomy and the other language taxonomies we need to remove the `pll_` prefix from their slug.

Reply:
Done in 2653e10cf16d0541ae4c6a66880ed433198155c9

Reply:
My future self thanks you 😁